File: web/routes/recommender.py
import socket
import sys
sys.path.append('/usr/lib/freecad/lib')

import FreeCAD
import Part
import ObjectsFem
import Fem
from femmesh import gmshtools
from femtools import ccxtools

import json

# ===== Create Socket Server =====
HOST, PORT = '127.0.0.1', 9999
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.bind((HOST, PORT))
s.listen(1)
print "Listening on PORT ", PORT

conn, addr = s.accept()
print "Client: ", addr

while True:
    data = conn.recv(1024)
    if not data:
        break

    print "Received: " + data
    conn.sendall(data)
conn.close()

File: web/routes/recommender.js
const net = require('net');
const stdin = process.openStdin();

let client = new net.Socket();
let PORT = 9999;

client.connect(PORT, '127.0.0.1', () => {
    console.log("Connected");
    stdin.addListener('data', (d) => {
	client.write(d.toString().trim());
    });
});

client.on('data', (data) => {
    console.log("Received: " + data);
});

client.on('close', () => {
    console.log("Disconnected from server");
});
